protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/timer/TimerRecord.java

```diff
@@ -99,4 +103,19 @@ public TimerRecord setWorkflowKey(long workflowKey) {
     this.workflowKeyProp.setValue(workflowKey);
     return this;
   }
+
+  public TimerType getTimerType() {
+    return this.timerType.getValue();
+  }
+
+  public TimerRecord setTimerType(TimerType type) {
+    this.timerType.setValue(type);
+    return this;
+  }
+
+  public enum TimerType {
+    CATCH,
+    START,
+    EVENT_SUBPROC,
+  }
```

Comment:
Currently, I don't see much benefit from adding the time type to the record. It is only used in `TriggerTimerProcessor` to write the EVENT_OCCURRED record a bit different. Compared to this, there are now many places where we pass or store the time type. So, the classes are now more coupled but they don't need to or should not (e.g. `ExecutableCatchEvent` from model package depends on the record).

We should try something different/simpler in the `TriggerTimerProcessor`.